settings/settings.go

```diff
@@ -227,3 +276,312 @@ func objectValidator(bound int, args ...string) bool {
+		}
+
+		settings.Airflow.Connections = append(settings.Airflow.Connections, newConnection)
+	}
+	// write to settings file
+	viperSettings.Set("airflow", settings.Airflow)
+	err = viperSettings.WriteConfig()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("successfully exported Connections\n\n")
+	return nil
+}
+
+func ExportVariables(id string, logs bool) error {
+	// setup files
+	airflowCommand := "airflow variables export tmp.var"
+	out := execAirflowCommand(id, airflowCommand)
+	if logs {
+		fmt.Println("Export Variables logs:\n" + out)
+	}
+
+	if strings.Contains(out, "successfully") {
+		// get variables created by the airflow command
+		fileCmd := "cat tmp.var"
+		out = execAirflowCommand(id, fileCmd)
+
+		m := map[string]string{}
+		err := json.Unmarshal([]byte(out), &m)
+		if err != nil {
+			fmt.Println("variable json decode unsuccessful")
+		}
+		// add the variables to settings object
+		for k, v := range m {
+			newVariables := Variables{{k, v}}
+			fmt.Println("Exporting Variable: " + k)
+			settings.Airflow.Variables = append(settings.Airflow.Variables, newVariables...)
+		}
+		// write variables to settings file
+		viperSettings.Set("airflow", settings.Airflow)
+		err = viperSettings.WriteConfig()
+		if err != nil {
+			return err
+		}
+		rmCmd := "rm tmp.var"
+		_ = execAirflowCommand(id, rmCmd)
+		fmt.Printf("successfully exported variables\n\n")
+		return nil
+	}
+	return errors.New("variable export unsuccessful")
+}
+
+func ExportPools(id string, logs bool) error {
+	// Setup airflow command to export pools
+	airflowCommand := "airflow pools list -o yaml"
+	out := execAirflowCommand(id, airflowCommand)
+	if logs {
+		fmt.Println("Export Pools logs:\n" + out)
+	}
+
+	// remove all color from output of the airflow command
+	plainOut := re.ReplaceAllString(out, "")
+
+	var pools ListPools
+	// remove warnings and extra text from the the output
+	yamlpools := "- description:" + strings.SplitN(plainOut, "- description:", 2)[1]
+
+	err := yaml.Unmarshal([]byte(yamlpools), &pools)
+	if err != nil {
+		return err
+	}
+	// add pools to the settings object
+	for i := range pools {
+		if pools[i].PoolName != "default_pool" {
+			slot, err := strconv.Atoi(pools[i].PoolSlot)
+			if err != nil {
+				fmt.Println("Issue with parsing pool slot number: ")
+				fmt.Println(err)
+			}
+			fmt.Println("Exporting Pool: " + pools[i].PoolName)
+			newPools := Pools{{pools[i].PoolName, slot, pools[i].PoolDescription}}
+			settings.Airflow.Pools = append(settings.Airflow.Pools, newPools...)
+		}
+	}
+	// write pools to the airflow settings file
+	viperSettings.Set("airflow", settings.Airflow)
+	err = viperSettings.WriteConfig()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("successfully exported pools\n\n")
+	return nil
+}
+
+func jsonString(connExtra map[interface{}]interface{}) string {
+	var extra_string string
+	i := 0
+	for k, v := range connExtra {
+		key, ok := k.(string)
+		value, ok := v.(string)
+		if ok {
```

Comment:
```suggestion
		key, keyOk := k.(string)
		value, valueOk := v.(string)
		if keyOk && valueOk {
```

nitpick: this might be the more correct way to express it